community/kernel/src/main/java/org/neo4j/kernel/impl/api/DefaultTransactionTracer.java

```diff
@@ -0,0 +1,183 @@
+    private volatile long totalTime = -1;
+
+    private final LogRotateEvent logRotateEvent = new LogRotateEvent()
+    {
+        @Override
+        public void close()
+        {
+            endTime = clock.currentTimeMillis();
+            totalTime = endTime - startTime;
+        }
+    };
+
+    private final LogAppendEvent logAppendEvent = new LogAppendEvent()
+    {
+        @Override
+        public void close()
+        {
+        }
+
+        @Override
+        public void setLogRotated( boolean logRotated )
+        {
+
+        }
+
+        @Override
+        public LogRotateEvent beginLogRotate()
+        {
+            startTime = clock.currentTimeMillis();
+            return logRotateEvent;
+        }
+
+        @Override
+        public SerializeTransactionEvent beginSerializeTransaction()
+        {
+            return SerializeTransactionEvent.NULL;
+        }
+
+        @Override
+        public LogForceWaitEvent beginLogForceWait()
+        {
+            return LogForceWaitEvent.NULL;
+        }
+
+        @Override
+        public LogForceEvent beginLogForce()
+        {
+            return LogForceEvent.NULL;
+        }
+    };
+
+    private final CommitEvent commitEvent = new CommitEvent()
+    {
+        @Override
+        public void close()
+        {
+        }
+
+        @Override
+        public LogAppendEvent beginLogAppend()
+        {
+            return logAppendEvent;
+        }
+
+        @Override
+        public StoreApplyEvent beginStoreApply()
+        {
+            return StoreApplyEvent.NULL;
+        }
+
+        @Override
+        public void setTransactionId( long transactionId )
+        {
+        }
+    };
+
+    private final TransactionEvent transactionEvent = new TransactionEvent()
+    {
+
+        @Override
+        public void setSuccess( boolean success )
+        {
+        }
+
+        @Override
+        public void setFailure( boolean failure )
+        {
+        }
+
+        @Override
+        public CommitEvent beginCommitEvent()
+        {
+            return commitEvent;
+        }
+
+        @Override
+        public void close()
+        {
+        }
+
+        @Override
+        public void setTransactionType( String transactionTypeName )
+        {
+        }
+
+        @Override
+        public void setReadOnly( boolean wasReadOnly )
+        {
+        }
+    };
+
+    public DefaultTransactionTracer()
+    {
+        this( Clock.SYSTEM_CLOCK );
+    }
+
+    public DefaultTransactionTracer( Clock clock )
+    {
+        this.clock = clock;
+    }
+
+    @Override
+    public TransactionEvent beginTransaction()
+    {
+        return transactionEvent;
+    }
+
+    @Override
+    public long lastLogRotationStartTime()
+    {
+        return startTime;
+    }
+
+    @Override
+    public long lastLogRotationEndTime()
+    {
+        return endTime;
+    }
+
+    @Override
+    public long lastLogRotationTotalTime()
+    {
+        return totalTime;
+    }
```

Comment:
For clarity I would append the unit to the method names, so they become `*TimeMillis`.